level/level.c

```diff
@@ -329,73 +333,116 @@ int rec_partition(int **room_map, int x, int y, int w, int h, int rm) {
 }
 
 static void partition(level *lvl) {
-    int **partitioning = malloc(lvl->width * sizeof(int*));
-    for (int i = 0; i != lvl->width; i++) partitioning[i] = malloc(lvl->height*sizeof(int));
+    // tile-to-room ID mapping
+    //TODO even clearer var name to remove this comment?
+    int **room_ids = malloc(lvl->width * sizeof(int*));
+    for (int i = 0; i != lvl->width; i++) {
+        room_ids[i] = malloc(lvl->height*sizeof(int));
+    }
+
     int **potential_doors= malloc(lvl->width * sizeof(int*));
-    for (int i = 0; i != lvl->width; i++) potential_doors[i] = malloc(lvl->height*sizeof(int));
+    for (int i = 0; i != lvl->width; i++) {
+        potential_doors[i] = malloc(lvl->height*sizeof(int));
+    }
 
-    int rooms = rec_partition(partitioning, 0, 0, lvl->width, lvl->height, 0);
+    // perform partitioning and get highest room ID
+    //TODO clearer function name to remove this comment
+    int max_room_id = rec_partition(room_ids, 0, 0, lvl->width, lvl->height, 0);
 
+    // initialize all tiles to bare floor
+    //TODO Optimization vs. readability: bare floor initialization
+    //TODO could be inside an 'else' down below.
     for (int x = 0; x < lvl->width; x++) {
         for (int y = 0; y < lvl->height; y++) {
             lvl->tiles[x][y] = TILE_FLOOR;
         }
     }
 
+    //TODO Any problem with skipping the border squares?
```

Comment:
Thoughts regarding skipping the border?

Reply:
The algorithm stops working? I mean, you could rewrite it to allow that, but why?

Reply:
Hmm... it didn't seem to. I guess I need to look at it more but removed the borders a while ago.

Reply:
Oh, oops. I am skipping the borders down below, not here.